tests/unit/tests_standard_lib/test_utilities/test_junit_parser.py

```diff
@@ -1,264 +1,72 @@
 
-def test_get_properties_raises_exception(junit_parser):
-    with pytest.raises(
-        Exception, match=" does not have all required properties"
-    ), patch.object(
-        JunitParser,
-        "yield_properties",
-        lambda x, tc: [
-            prop(name="data_model", value=f"data_model"),
-            prop(name="data_set", value=f"data_set"),
-            prop(name="fields", value=f"fields"),
-            prop(name="tag_stanza", value=f"tag_stanza"),
-            prop(name="fake_prop", value=f"fake_prop"),
-        ],
-    ):
-        junit_parser.get_properties(
-            testcase(name="test_cim_required_fields", result=None)
+def test_parse_junit_raises_exception_when_not_fields_are_present_in_testcase():
+    jp = JunitParser(
+        os.path.join(
+            os.path.dirname(__file__), "test_data", "sample_junit_without_datamodel.xml"
         )
-
-
-def test_yield_properties(junit_parser):
-    tc = MagicMock()
-    tc.child.side_effect = lambda x: (
-        item
-        for item in [
-            "data_model",
-            "data_set",
-            "fields",
-            "tag_stanza",
-            "fields_type",
-            "fake_prop",
-        ]
     )
-    out = list(junit_parser.yield_properties(tc))
-    assert out == [
-        "data_model",
-        "data_set",
-        "fields",
-        "tag_stanza",
-        "fields_type",
-        "fake_prop",
+    with pytest.raises(Exception):
+        jp.parse_junit()
+
+
+def test_junit_parser_parser_junit():
+    expected_result = [
+        {
+            "data_model": "datamodel_1",
+            "fields": "field_1",
+            "data_set": "dataset_1",
+            "fields_type": "required",
+            "tag_stanza": 'dataset_name="dataset_1"',
+            "status": "failed",
+            "test_property": "splunklib.binding.HTTPError: HTTP 400 Bad Request",
+        },
+        {
+            "data_model": "datamodel_2",
+            "fields": "field_2",
+            "data_set": "dataset_2",
+            "fields_type": "required",
+            "tag_stanza": 'dataset_name="dataset_2"',
+            "status": "failed",
+            "test_property": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        },
+        {
+            "data_model": "datamodel_3",
+            "fields": "field_3",
+            "data_set": "dataset_3",
+            "fields_type": "conditional",
+            "tag_stanza": 'dataset_name="dataset_3"',
+            "status": "passed",
+            "test_property": "-",
+        },
+        {
+            "data_model": "datamodel_4",
+            "fields": "field_4",
+            "data_set": "dataset_4",
+            "fields_type": "required",
+            "tag_stanza": 'dataset_name="dataset_4"',
+            "status": "skipped",
+            "test_property": "-",
+        },
     ]
+    jp = JunitParser(
+        os.path.join(os.path.dirname(__file__), "test_data", "sample_junit.xml")
```

Comment:
It looks like the test should load .../test_utilities/test_data/sample_junit.xml, parse it and compare with expected result, correct? I see above that sample_junit.xml is empty so test should probably fail.

Reply:
Github shows it as "File renamed without changes.", I just moved that file to another directory.

Reply:
Yes, sorry, your are right, I misunderstood the 0 value